### ISSUE TITLE
feat: add the `clear` feature

### ISSUE THREAD

**Issue by iqdecay** [FuelLabs/forc-wallet#8]
```bash
# Clear a wallet from storage.
forc wallet clear <wallet number>
```

**Comment by adlerjohn** [comment]
What does this specifically entail? HD wallets are derived, not stored. Clearing a derived key doesn't actually do anything.

**Comment by kayagokalp** [comment]
I think with `new` command's implementation at #3, after randomly generating the wallet we are encrypting and storing it in the vault ([here](https://github.com/FuelLabs/forc-wallet/blob/e52d67cb0d74e636639a9ec5acdbd47e9c55d0dc/src/main.rs#L94) like [fuels-doc example](https://fuellabs.github.io/fuels-rs/v0.20.0/wallets/encrypting-and-storing.html#encrypting-and-storing-a-wallet-created-from-a-mnemonic-or-private-key)). My understanding was that `clear` should be removing that from the vault (basically deleting corresponding encrypted file/folder from the disk) since this is first documented in #2 maybe @digorithm might have a better input 🙌

**Comment by adlerjohn** [comment]
> after randomly generating the wallet we are encrypting and storing it in the vault

That needs to be spec'd out first. Is the wallet going to use a mnemonic? Will `forc-wallet new` generate a new mnemonic?

**Comment by adlerjohn** [comment]
There has to be a separate issue to specify `forc-wallet new`.

**Comment by digorithm** [comment]
> > after randomly generating the wallet we are encrypting and storing it in the vault
> 
> 
> 
> That needs to be spec'd out first. Is the wallet going to use a mnemonic? Will `forc-wallet new` generate a new mnemonic?

Yes and yes -- that's what's happening, at least in the PoC version (https://github.com/FuelLabs/forc-wallet/pull/3). 

Mnemonics are generated for the user and stored in an encrypted vault (see the gif in the linked PR if you can't go through the code at the moment). 

Of course, it's all malleable; the PoC is just a PoC 🙂

**Comment by kayagokalp** [comment]
Should we close this as this is not relevant or do we need a command for removing the encrypted phrase from the disk (Which currently happens while initializing a new HD wallet)?

**Comment by adlerjohn** [comment]
It's not immediately relevant, but it could be relevant later if the utility keeps track of additional data for each account, e.g. a cache of past actions.
